packages/subgraph/src/packages/token/governance-erc20.ts

```diff
@@ -5,72 +5,157 @@ import {
+import {ADDRESS_ZERO} from '../../utils/constants';
 import {generateMemberEntityId} from '../../utils/ids';
-import {Address, BigInt, dataSource, store} from '@graphprotocol/graph-ts';
+import {Address, BigInt, dataSource, log, store} from '@graphprotocol/graph-ts';
 
-function getOrCreateMember(user: Address, pluginId: string): TokenVotingMember {
+function getERC20Balance(user: Address, tokenAddress: Address): BigInt {
+  let contract = GovernanceERC20Contract.bind(tokenAddress);
+  let balance = contract.balanceOf(user);
+  return balance;
+}
+
+function getDelegation(user: Address, tokenAddress: Address): string | null {
+  let contract = GovernanceERC20Contract.bind(tokenAddress);
+  let delegate = contract.delegates(user);
+
+  return delegate === Address.fromString(ADDRESS_ZERO)
+    ? null
+    : delegate.toHexString();
+}
+
+function getDelegateeId(
+  user: Address,
+  tokenAddress: Address,
+  pluginId: string
+): string | null {
+  let delegatee = getDelegation(user, tokenAddress);
+  return delegatee
+    ? generateMemberEntityId(
+        Address.fromString(pluginId),
+        Address.fromString(user.toHexString())
+      )
+    : null;
+}
+
+function getVotingPower(user: Address, tokenAddress: Address): BigInt {
+  let contract = GovernanceERC20Contract.bind(tokenAddress);
+  let votingPower = contract.getVotes(user);
+  return votingPower;
+}
+
+/**
+ * A container for the result of the `getOrCreateMember` function.
+ * @param entity - The `TokenVotingMember` entity.
+ * @param createdNew - A boolean indicating whether the entity was created new
+ * or if false it was previously created. If the entity was created new, it already
+ * has the latest balance of the user, so no need to then update the balance.
+ */
+class TokenVotingMemberResult {
+  entity: TokenVotingMember;
+  createdNew: boolean;
+
+  constructor(entity: TokenVotingMember, createNew: boolean) {
+    this.entity = entity;
+    this.createdNew = createNew;
+  }
+}
+
+function getOrCreateMember(
+  user: Address,
+  pluginId: string,
+  tokenAddress: Address
+): TokenVotingMemberResult {
   let memberEntityId = generateMemberEntityId(
-    user,
-    Address.fromString(pluginId)
+    Address.fromString(pluginId),
+    user
   );
+  let createdNew = false;
   let member = TokenVotingMember.load(memberEntityId);
+
   if (!member) {
+    createdNew = true;
     member = new TokenVotingMember(memberEntityId);
     member.address = user;
-    member.balance = BigInt.zero();
+    member.balance = getERC20Balance(user, tokenAddress);
     member.plugin = pluginId;
 
-    member.delegatee = null;
-    member.votingPower = BigInt.zero();
+    member.delegatee = getDelegateeId(user, tokenAddress, pluginId);
+    member.votingPower = getVotingPower(user, tokenAddress);
   }
 
-  return member;
+  return new TokenVotingMemberResult(member, createdNew);
 }
 
 export function handleTransfer(event: Transfer): void {
```

Comment:
since this is a mapping file, 
all the lines above this handler are helper functions,
it would have been nice to move those helpers to an utils file in the same folder similar to what we do in the rest of subgraph,
WDYT?

Reply:
Yes makes sense.